surfactant/cmd/generate.py

```diff
@@ -431,10 +431,16 @@ def sbom(
                                 except Exception as e:
                                     raise RuntimeError(f"Unable to process: {filepath}") from e
 
+                        if not entry.includeFileExts:
+                            entry.includeFileExts = []
+                        if not entry.excludeFileExts:
+                            entry.excludeFileExts = []
+
                         if (
                             ftype := pm.hook.identify_file_type(filepath=filepath)
                             or include_all_files
-                        ):
+                            or os.path.splitext(filepath)[1] in entry.includeFileExts
+                        ) and os.path.splitext(filepath)[1] not in entry.excludeFileExts:
```

Comment:
```suggestion
                            or filepath.suffix in entry.includeFileExts
                        ) and filepath.suffix not in entry.excludeFileExts:
```

`filepath` should be of type `pathlib.Path`, so we can use `.suffix` to get the extension.

Reply:
Nevermind, looks like `filepath` is set from a call to `as_posix()` so its actually a `str`.